src/main/scala/viper/server/vsi/HttpVerificationServerInterface.scala

```diff
@@ -0,0 +1,135 @@
+package viper.server.vsi
+
+import akka.actor.PoisonPill
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.NotUsed
+import akka.pattern.ask
+import akka.stream.scaladsl.Source
+import akka.util.Timeout
+
+import spray.json.{DefaultJsonProtocol, RootJsonFormat}
+
+import viper.server.vsi.VerificationProtocol.Stop
+
+import scala.concurrent.duration._
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+
+/** This expands a VerifiationServer (VerSer) by providing common HTTP functionality.
+  *  (E.g.: Stopping a VerSer, submitting Verification Requests and requesting their results)
+  *
+  * The VerSer- specific functionality will need to be implemented for each individual server.
+  * In particular, this means providing a protocol that returns the VerSer's responses as type
+  * [[ToResponseMarshallable]].
+  * */
+trait HttpVerificationServerInterface extends VerificationServerInterface
```

Comment:
In Scala, one normally doesn't have to add a new train for each combination of traits that one wants to lated mix into one's classes. Scala permits multiple inheritance via mixing in many traits at the same time (note that the order of mixed-in traits matters when the traits being mixed in specify fields or methods with signatures that cannot be overloaded; you can find more info about this in the docs). 

That being said, is it still needed to have this trait? It would be more convenient to have smaller building blocks, i.e., independent traits.

Reply:
> In Scala, one normally doesn't have to add a new train for each combination of traits that one wants to lated mix into one's classes.

As mentioned in the comment in the HttpServerInterface file this mixing of traits is not necessary.

> That being said, is it still needed to have this trait? It would be more convenient to have smaller building blocks, i.e., independent traits.

I'm not sure I understand. This trait is really the meat of providing an HTTP interface for a verification server. I came to the conclusion that these routes are essential to have in any verification server and therefore require implementation by a client.